fix MPI compilation issues by not requiring that CLANG_UNDER_VS preprocessor symbol is defined to include mpi.h
author: Kenneth Hoste (HPC-UGent)
--- iqtree2-2.2.1/pll/genericParallelization.h.orig	2022-06-01 18:11:42.563407234 +0200
+++ iqtree2-2.2.1/pll/genericParallelization.h	2022-06-01 18:10:02.554264000 +0200
@@ -63,7 +63,7 @@
 /******************/
 /* MPI SPECIFIC   */
 /******************/
-#if defined(_FINE_GRAIN_MPI) || (defined(_IQTREE_MPI) && defined(CLANG_UNDER_VS))
+#if defined(_FINE_GRAIN_MPI) || defined(_IQTREE_MPI)
 #include <mpi.h>
 #ifdef DEBUG_MPI_EACH_SEND
 #define DEBUG_PRINT(text, elem) printf(text, elem)
